### PR TITLE
Add Matrix social icon

### PR DESCRIPTION
**Pull request by orhun**
#### Description

- Closes #1155

### CODE REVIEW

**Review by at-the-vr — APPROVED** [review]
Nicely Done, Matrix icon looks perfect [here](https://starlight-oeb6jbr8f-astrodotbuild.vercel.app/guides/components/#all-icons)

File: packages/starlight/components/Icons.ts [hunk 1]
```diff
@@ -101,4 +101,6 @@ export const Icons = {
...
+	matrix:
+		'<path d="M22.5 1.5v21h-2.25V24H24V0h-3.75v1.5h2.25ZM7.46 7.95V9.1h.04a3.02 3.02 0 0 1 2.61-1.39c.54 0 1.03.1 1.48.32.44.2.78.58 1.01 1.1.26-.37.6-.7 1.03-.99.44-.28.95-.43 1.55-.43.45 0 .87.06 1.26.17.38.11.71.29.99.53.27.24.49.56.64.95.15.4.23.86.23 1.42v5.72h-2.34v-4.85c0-.29-.01-.56-.04-.8a1.73 1.73 0 0 0-.18-.67 1.1 1.1 0 0 0-.44-.45 1.6 1.6 0 0 0-.78-.16c-.33 0-.6.06-.8.19-.2.12-.37.29-.48.5a2 2 0 0 0-.23.69c-.04.26-.06.52-.06.78v4.77H10.6v-4.8l-.01-.75a2.29 2.29 0 0 0-.14-.69c-.08-.2-.23-.38-.42-.5a1.5 1.5 0 0 0-.85-.2c-.15.01-.3.04-.44.08-.19.06-.37.15-.52.28-.18.14-.32.34-.44.6-.12.26-.18.6-.18 1.02v4.96H5.25V7.94h2.21ZM1.5 1.5v21h2.25V24H0V0h3.75v1.5H1.5Z"/>'
```
Comment (at-the-vr) [inline packages/starlight/components/Icons.ts:105]: Small suggestion (can have trailing `,` at the end as seen in previous commits)
```suggestion
		'<path d="M22.5 1.5v21h-2.25V24H24V0h-3.75v1.5h2.25ZM7.46 7.95V9.1h.04a3.02 3.02 0 0 1 2.61-1.39c.54 0 1.03.1 1.48.32.44.2.78.58 1.01 1.1.26-.37.6-.7 1.03-.99.44-.28.95-.43 1.55-.43.45 0 .87.06 1.26.17.38.11.71.29.99.53.27.24.49.56.64.95.15.4.23.86.23 1.42v5.72h-2.34v-4.85c0-.29-.01-.56-.04-.8a1.73 1.73 0 0 0-.18-.67 1.1 1.1 0 0 0-.44-.45 1.6 1.6 0 0 0-.78-.16c-.33 0-.6.06-.8.19-.2.12-.37.29-.48.5a2 2 0 0 0-.23.69c-.04.26-.06.52-.06.78v4.77H10.6v-4.8l-.01-.75a2.29 2.29 0 0 0-.14-.69c-.08-.2-.23-.38-.42-.5a1.5 1.5 0 0 0-.85-.2c-.15.01-.3.04-.44.08-.19.06-.37.15-.52.28-.18.14-.32.34-.44.6-.12.26-.18.6-.18 1.02v4.96H5.25V7.94h2.21ZM1.5 1.5v21h2.25V24H0V0h3.75v1.5H1.5Z"/>',
```

**Comment by at-the-vr** [comment]
bumping this up to reach Hideoo 😁

**Review by HiDeoo — APPROVED** [review]
Thanks for your contribution :raised_hands: 

Just tested locally and this looks great!

Once reviewed by Chris, we will wait for the next release to merge this as merging this will automatically update the documentation to mention that this new icon is available and we don't want to do that too early before the actual release :thumbsup:

File: .changeset/strange-boxes-know.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+'@astrojs/starlight': patch
+---
+Add Matrix social link icon
```
[the remark this answers is not in the document]
Comment (orhun) [reply .changeset/strange-boxes-know.md]: Done in e75c79c71c8eb2915fece836e5b8a6f22e541ada

**Review by ElianCodes — APPROVED** [review]
Looking fancy!

**Review by delucis — APPROVED** [review]
Thanks again for this @orhun! Time to release the matrix 🥳

**Comment by orhun** [comment]
We now have a Matrix icon on our website: https://ratatui.rs

Thanks everyone! 🧀

**Comment by HiDeoo** [comment]
> We now have a Matrix icon on our website: [ratatui.rs](https://ratatui.rs)
> 
> Thanks everyone! 🧀

Thanks you for your contribution 🙌 (and the Ratatui  website looks amazing 🤩 )

**Pull request merged**